src/hooks/usePodOwner/index.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright 2020 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import { useState, useEffect } from "react";
+import { getResourceInfo, getPodOwner } from "@inrupt/solid-client";
+import { useSession } from "@inrupt/solid-ui-react";
+
+export default function usePodOwner(resourceIri) {
+  const { fetch } = useSession();
+  const [podOwnerWebId, setPodOwnerWebId] = useState(null);
+  const [error, setError] = useState();
+
+  useEffect(() => {
+    if (!resourceIri) return;
```

Comment:
You should set both `error` and `podOwnerWebId` to null in this case, as there are cycles where resourceIri will be undefined by design (e.g. the caller needs to load something before resourceIri is ready), and we don't want to use older instances of `error` and `podOwnerWebId` in those cases.